GoogleForCreators/web-stories-wp
# Dashboard: Display error messages in UI

## Summary
Creates a new view component for the dashboard to display errors as toasts for stories and templates. 

## Relevant Technical Choices
- Revises `error` object to have a message.title and message.body to give context to displayed toasts. Adds `Date.now()` as a unique ID that will return in ascending order to maintain toast stack when there are many errors to display at once. 
- Mocking Date in reducer tests to get Date.now() equality. 

## User-facing changes
- Now when APIs return errors for the below actions you will see them in the UI:
  - Updating story
  - Deleting story
  - Duplicating story
  - Fetching stories
  - Creating story from template
  - Creating a template from a story

## Testing Instructions
- Storybook: Dashboard/Views/Toaster 
- Otherwise, you need to cause API failures. One way I have been testing this is the following:
  - view the dashboard, open your dev tools and go to the network tab. Right click on one of the 'web-story?' xhr requests and click 'block url' - repeat the action that triggered that request and see an error message show up. 
  - Alternatively you can always block the domain, but I find that to be a little noisy and harder to pinpoint what's going on. 

**unable to delete story**
![unable to delete story](https://user-images.githubusercontent.com/10720454/85322015-7e86e180-b47a-11ea-805d-d970b1005a27.gif)

**unable to update story**
![unable to update story](https://user-images.githubusercontent.com/10720454/85322027-847cc280-b47a-11ea-939a-7ffee032d2c7.gif)

**toaster consistency in grid to list**
![toaster consistency in grid to list](https://user-images.githubusercontent.com/10720454/85322035-86df1c80-b47a-11ea-9ead-0432c1f24d4e.gif)

**unable to load stories**
![unable to load stories](https://user-images.githubusercontent.com/10720454/85322045-8cd4fd80-b47a-11ea-8bc2-d95b5df11e8f.gif)


---



Addresses #2262
